src/snarky.d.ts

```diff
@@ -51,489 +53,806 @@ declare interface ProvablePure<T> extends Provable<T> {
+   * 
+   * @return A number representing the size of the {@link Field} type in terms of {@link Field} type itself.
    */
   static sizeInFields(): number;
 
   /**
-   * Static method to serialize a {@link Field} into an array of {@link Field} elements.
+   * This function is the implementation of {@link Provable.toFields} in {@link Field} type.
+   * Static function to serializes a {@link Field} into an array of {@link Field} elements.
+   * You can use this array to calculate the {@link Poseidon} hash of a {@link Field}.
+   * This will be always an array of length 1, where the first and only element equals the given parameter itself.
+   * 
+   * @param value - the {@link Field} element to cast the array from.
+   * 
+   * @return A {@link Field} array of length 1 created from this {@link Field}.
    */
-  static toFields(x: Field): Field[];
+  static toFields(value: Field): Field[];
+
   /**
-   * Static method to serialize a {@link Field} into its auxiliary data.
+   * This function is the implementation of {@link Provable.toAuxiliary} in {@link Field} type.
+   * As the primitive {@link Field} type has no auxiliary data associated with it, this function will always return an empty array.
+   * 
+   * @param value - The {@link Field} element to get the auxiliary data of, optional. If not provided, the function returns an empty array.
    */
-  static toAuxiliary(x?: Field): [];
+  static toAuxiliary(value?: Field): [];
 
-  /*
-  static assertEqual(
-    x: Field | number | string | boolean,
-    y: Field | number | string | boolean
-  ): Field;
-  static assertBoolean(x: Field | number | string | boolean): void;
-  static isZero(x: Field | number | string | boolean): Bool;
-  */
+  /**
+   * Convert a bit array into a {@link Field} element using [little endian binary representation](https://en.wikipedia.org/wiki/Endianness)
+   * The function fails if the element cannot fit given too many bits. Note that a {@link Field} element can be 254 bits at most.
+   * 
+   * **Important: If the given `bytes` array is an array of `booleans` or {@link Bool} elements that all are `constant`, the resulting {@link Field} element will be a constant as well. Or else, if the given array is a mixture of constants and variables of {@link Bool} type, the resulting {@link Field} will be a variable as well.
+   * 
+   * @param bytes - An array of {@link Bool} or `boolean` type.
+   * 
+   * @return A {@link Field} element matching the [little endian binary representation](https://en.wikipedia.org/wiki/Endianness) of the given `bytes` array.
+   */
+  static fromBits(bytes: (Bool | boolean)[]): Field;
 
   /**
-   * Converts a bit array into a field element (little endian)
-   * Fails if the field element cannot fit given too many bits.
+   * Serialize the given {@link Field} element to a JSON string, e.g. for printing. If you try to print a {@link Field} without this function it will directly stringify the Field object, resulting in an unreadable output.
```

Comment:
same as in https://github.com/o1-labs/snarkyjs/pull/917/files#diff-c986331433c1c055aa5d451d0b789e5f5e9aee97f754e6f1550fb09b5acc91b4R321